crates/matrix-sdk-sqlite/src/event_cache_store.rs

```diff
@@ -185,10 +194,208 @@ impl EventCacheStore for SqliteEventCacheStore {
 
     async fn handle_linked_chunk_updates(
         &self,
-        _room_id: &RoomId,
-        _updates: Vec<Update<Event, Gap>>,
+        room_id: &RoomId,
+        updates: Vec<Update<Event, Gap>>,
     ) -> Result<(), Self::Error> {
-        todo!()
+        let hashed_room_id = self.encode_key(keys::LINKED_CHUNKS, room_id);
+
+        for up in updates {
+            match up {
+                Update::NewItemsChunk { previous, new, next } => {
+                    let hashed_room_id = hashed_room_id.clone();
+
+                    let previous = previous.as_ref().map(ChunkIdentifier::index);
+                    let new = new.index();
+                    let next = next.as_ref().map(ChunkIdentifier::index);
+
+                    trace!(
+                        %room_id,
+                        "new events chunk (prev={previous:?}, i={new}, next={next:?})",
+                    );
+
+                    self.acquire()
+                        .await?
+                        .with_transaction(move |txn| {
+                            insert_chunk(txn, &hashed_room_id, previous, new, next, "E")
```

Comment:
> I think it's much more robust and can detect errors at compile-time instead of at runtime with SQLite.

Not really, since we'd still need a `TryFrom<&str> for ChunkKind`, for reading from the database, so there's still a possible runtime error *and* we now have an extra data structure which adds cognitive overhead? I can add constants for the strings, though, to avoid scattering :+1: